### PR TITLE
fix flannel setup on 1.6 cluster

### LINKED ISSUE #2284: Master status NotReady with --networking=flannel or weave on kops 1.6 alpha and kubernetes 1.6
Flannel pod is not getting scheduled in the master node, so it stays in NotReady state. It seems to be working fine for other nodes though

### PR DESCRIPTION
**Pull request by felipejfc**
fixes #2284 @justinsb @chrislovecnm

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2292)

### CODE REVIEW

File: upup/pkg/fi/cloudup/bootstrapchannelbuilder.go [hunk 1]
```diff
@@ -226,8 +226,13 @@ func (b *BootstrapChannelBuilder) buildManifest() (*channelsapi.Addons, map[stri
...
 	if b.cluster.Spec.Networking.Flannel != nil {
 		key := "networking.flannel"
-		version := "0.7.0"
-
+		var version string
+		switch {
+		case kv.Major == 1 && kv.Minor <= 5:
+			version = "0.7.0"
+		default:
+			version = "0.7.0_1.6"
```
Comment (justinsb) [inline upup/pkg/fi/cloudup/bootstrapchannelbuilder.go]: I'm trying to figure out how to deal with this... I think we need to reinstall whenever the k8s version changes, which will require some changes to the infrastructure.  Sorry for the delay.

**Comment by justinsb** [comment]
So I think I've figured out how to do this now ... we can have a `kubernetesVersion` selector, and if you want to continue to upgrade the addon for older kubernetes versions, we have an `id` field which solves the problem of upgrading and downgrading.  That work is in PR #2314, but the docs are here: https://github.com/justinsb/kops/blob/e8921862f636de3003534491f1dd94fdc32682c1/docs/addon_manager.md

I'll probably merge this and convert it to the new approach, but if you beat me to it ... :-)

**Comment by justinsb** [comment]
@felipejfc any chance you could rebase this so I can merge it; I will then follow up with a change that does the same thing as this: https://github.com/kubernetes/kops/pull/2337

Or if you want to do the same change, you're 99% of the way there already :-)

**Comment by felipejfc** [comment]
I'll do it @justinsb, give me a couple of hours 👊🏼

**Comment by felipejfc** [comment]
@justinsb can you see it there's anything wrong with my last push? I rebased my branch with master and made the changes you've asked but creating a 1.6.1 and a 1.5.5 both used networking.flannel/pre-k8s-1.6.yaml

**Comment by justinsb** [comment]
Looks great - thanks @felipejfc!  I think the problem was just around uploading a newer version of nodeup/protokube/channels

**Pull request merged**